irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -93,37 +94,37 @@ namespace iroha {
                 return {};
               }
               return expected::makeError(
-                  formSignaturesErrorMsg(tx.signatures(), signatories));
+                  this->formSignaturesErrorMsg(tx.signatures(), signatories));
             });
       };
 
       // Filter only valid transactions and accumulate errors
       auto transactions_errors_log =
-          std::vector<std::pair<std::vector<std::string>, int>>{};
+          std::vector<std::pair<std::vector<std::string>, size_t>>{};
       auto filter = [&temporaryWsv,
                      checking_transaction,
-                     &transactions_errors_log](auto &tx, int tx_index) {
+                     &transactions_errors_log,
+                     &proposal](auto &tx) {
         return temporaryWsv.apply(tx, checking_transaction)
-            .match([](expected::Value<void> &) { return true; },
-                   [&transactions_errors_log, tx_index](
-                       expected::Error<std::vector<std::string>> &error) {
-                     transactions_errors_log.push_back(
-                         std::make_pair(error.error, tx_index));
-                     return false;
-                   });
+            .match(
+                [](expected::Value<void> &) { return true; },
+                [&transactions_errors_log,
+                 &proposal, &tx](expected::Error<std::vector<std::string>> &error) {
+                  auto txs = proposal.transactions();
+                  transactions_errors_log.push_back(std::make_pair(
+                      error.error, boost::range::find(txs, tx) - txs.begin()));
```

Comment:
Still, please consider using `boost::adaptors::filtered` and `boost::adaptors::indexed` to avoid linear `find`.

Reply:
The problem is that there's no way one can pass additional arguments to ```filtered```, I tried to use bind or lambda capture, but failed